elastic/beats
# Feature/cache replacament

## What does this PR do?

Replaced time expiring cache in the Kubernetes module with a dictionary that never expires. 

Since some metrics in this internal cache were only set once at startup and not regularly updated, we needed to replace the internal cache with a dictionary that never expires.

## Why is it important?

It fixes an issue that caused:
- if state_node, node metricsets are disabled, 
  - container with limit sets
    - accurate metrics before cache expiration
    - missing metrics after cache expiration
  - container without limit sets
    - missing metrics. This is CORRECT since limits are not set on containers and we cannot use node metrics as backup option
  - pod with container limits sets
    - missing metrics 
  - pod without limit sets
    - missing metrics. This is CORRECT since limits are not set on containers and we cannot use node metrics as backup option
- if state_node, node metricsets are enabled,
  - container/pod with limit sets
    - accurate metrics before cache expiration
    - missing metrics after cache expiration
  - container/pod without limit sets
    - inaccurate metrics before cache expiration.  This is CORRECT since limits are not set on containers and we use node metrics as backup option
    - missing metrics after cache expiration

## Checklist


## Author's Checklist

- if state_node, node metricsets are disabled, 
  - container/pod with limit sets
    - accurate metrics
  - container/pod without limit sets
    - missing metrics. This is CORRECT since limits are not set on containers and we cannot use node metrics as backup option
- if state_node, node metricsets are enabled,
  - container/pod with limit sets
    - accurate metrics 
  - container/pod without limit sets
    - inaccurate metrics. This is CORRECT since limits are not set on containers and we use node metrics as backup option

## How to test this PR locally

- Run metricbeat on kubernetes with either Kind with multiple nodes or a k8s cluster on GCP/AWS
- Add/remove test pods and see metrics appear/disappear for those pods
- check logs in debug mode for entries added/removed from the metricsStorage dictionary

## Related issues


- 

## Use cases



## Screenshots



## Logs
